When hypothesis is available on the system, the Python test suite incorrectly
assumes it's running in GitHub CI and tries to download data from there,
resulting in several failing tests.

This was reported and fixed upstream:

https://github.com/python/cpython/issues/132316
https://github.com/python/cpython/pull/132367

From f682e9edcebb05f349788fc9546c57ea1fe29664 Mon Sep 17 00:00:00 2001
From: sobolevn <mail@sobolevn.me>
Date: Thu, 10 Apr 2025 14:30:58 +0300
Subject: [PATCH] gh-132316: Require `socket` and `GITHUB_TOKEN` env to use
 `GitHubArtifactDatabase` (GH-132348) (cherry picked from commit
 fcf2d07228ada64bc864380cd8e3e304d9768caa)

Co-authored-by: sobolevn <mail@sobolevn.me>
Co-authored-by: Hugo van Kemenade <1324225+hugovk@users.noreply.github.com>

--- Python-3.13.3/Lib/test/support/hypothesis_helper.py
+++ Python-3.13.3/Lib/test/support/hypothesis_helper.py
@@ -5,6 +5,7 @@ try:
 except ImportError:
     from . import _hypothesis_stubs as hypothesis
 else:
+    from test.support import has_socket_support
     # When using the real Hypothesis, we'll configure it to ignore occasional
     # slow tests (avoiding flakiness from random VM slowness in CI).
     hypothesis.settings.register_profile(
@@ -21,7 +22,14 @@ else:
     # of failing examples, and also use a pull-through cache to automatically
     # replay any failing examples discovered in CI.  For details on how this
     # works, see https://hypothesis.readthedocs.io/en/latest/database.html
-    if "CI" not in os.environ:
+    # We only do that if a GITHUB_TOKEN env var is provided, see:
+    # https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens
+    # And Python is built with socket support:
+    if (
+        has_socket_support
+        and "CI" not in os.environ
+        and "GITHUB_TOKEN" in os.environ
+    ):
         from hypothesis.database import (
             GitHubArtifactDatabase,
             MultiplexedDatabase,
